backend/user.js

```diff
@@ -550,21 +556,33 @@ router.put('/appointments/:id', authenticateToken, async (req, res) => {
     });
 
     // Find the patient associated with this appointment
-    const patient = await findPatient(req.user.id);
-    // Create a new notification
-    const newNotification = await prisma.notification.create({
-      data: {
-        content: `Updated Appointment: ${title} on ${date}`,
-        patientId: patient.id,
-      },
+    const appointment = await prisma.appointment.findUnique({
+      where: { id: parseInt(id, 10) },
+      include: { patient: true }
     });
 
-    // Send notification via WebSocket
-    if (userToWS[patient.userId]) {
-      userToWS[patient.userId]({
-        message: `Updated Appointment: ${title} on ${date}`,
-        isNotification: true,
+    if (!appointment || !appointment.patient) {
+      return res.status(404).json({ error: 'Patient profile not found' });
+    }
+
+    const patient = appointment.patient;
+
+    if (patient.notificationsOn){
+      // Create a new notification
+      const newNotification = await prisma.notification.create({
+        data: {
+          content: `Updated Appointment: ${title} on ${date}`,
```

Comment:
Can we modularize the process of creating a notif + sending the message via websocket with a helper function? Seems like we had to repeat it ~3 times in this PR